### DISCUSSION TITLE
round_end event not being triggered correctly

### DISCUSSION THREAD

**Discussion by mavrick** [saul/demofile#227] [Questions] [answered]
![image](https://user-images.githubusercontent.com/357659/36025672-937fbb64-0ddf-11e8-8652-5bd8cd0def1e.png)

i'm only seeing a total of 20 ``round_end`` events for match 2 that should have 26.

Match Data: https://www.hltv.org/matches/2318632/faze-vs-cloud9-eleague-major-2018

All 3 demos from the GOTV are missing ``round_end`` events.

**match 1:** https://mega.nz/#!OwFC2Q6J!mye2CCLGE8xojqDd6h6gy2aXPrH5kkfRSEIcUtlXtgI
**match 2:** https://mega.nz/#!ihURHbxK!IumMuyWguCWg4DKmj8jKFPyDTq0wyrRolN_JjmRcnpA
**match 3:** https://mega.nz/#!e9sWEJSD!9WYbyWozXZOZ7Ftxpv01G_S5NBx12paJz_ysHQr71wc

But if I process a match from IEM Oakland https://www.hltv.org/matches/2317279/sk-vs-astralis-iem-oakland-2017

The ``round_end`` events are correct.

Using latest version of ``demofile``

**Comment by mavrick** [comment]
Looks like most if not all ELEAGUE 2018 demos are missing data or are not triggering events correctly

https://www.hltv.org/matches/2318597/cloud9-vs-g2-eleague-major-2018

Only 16 ``round_end`` events trigger for this demo

**Comment by mavrick** [comment]
Looking at the CSGO Demo Manager code there are comments about ``round_end`` 

```
/// Sometimes round_end event isn't triggered, this variable is used to handle this case and update scores at round_officialy_end event
```
```
/// Between round_end and round_officialy_ended events may happened so we update data at round_officialy_end
/// However at the last round of a match, round officially end isn't raised (on Valve demos)
```

**Comment by saul** [comment]
Looks like this has nothing to do with the parser - https://github.com/ValveSoftware/csgo-demoinfo/issues/20, not much I can do here.

Is it not possible for you to use round_officially_ended instead?

**Comment by mavrick** [comment]
Yes but we don't seem to get an event information like reason, winner, message, etc but I can calculate the winner based on the team scores easily enough.

**Comment by saul** [comment, marked as answer]
Yeah I think the problem here is that the event isn’t sent reliably (the engine has support for this), so if the packet is lost then the event isn’t recorded by the GOTV relays
